R/col_types.R

```diff
@@ -82,30 +82,50 @@ as.col_spec.default <- function(x) {
 }
 
 #' @export
-print.col_spec <- function(x, n = Inf, ...) {
-  cat(format.col_spec(x, n = n, ...))
+print.col_spec <- function(x, n = Inf, condense = FALSE, ...) {
+  cat(format.col_spec(x, n = n, condense = condense, ...))
 
   invisible(x)
 }
 
+#' @rdname spec
 #' @export
-format.col_spec <- function(x, n = Inf, ...) {
+condense_spec <- function(x) {
+  types <- vapply(x$cols, function(xx) class(xx)[[1]], character(1))
+  counts <- table(types)
+  most_common <- names(counts)[counts == max(counts)][[1]]
+
+  x$default <- x$cols[types == most_common][[1]]
+  x$cols <- x$cols[types != most_common]
+  x
+}
+
+#' @export
+format.col_spec <- function(x, n = Inf, condense = FALSE, ...) {
```

Comment:
I think your suggestion is better, previously there was no way to explicitly force the output to be condensed (other than calling condense_spec()) directly.